bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineFactory.java

```diff
@@ -14,24 +14,30 @@
 
 import java.util.List;
 import java.util.Map;
+import java.util.TreeMap;
 
 import javax.script.ScriptEngine;
 
 /**
  * This class is used by the ScriptManager to load ScriptEngines.
  * This is meant as a way to allow other OSGi bundles to provide custom Script-Languages with special needs (like
- * Nashorn, Groovy, etc.)
+ * Nashorn, Jython, Groovy, etc.)
  *
  * @author Simon Merschjohann
- *
+ * @author Scott Rushworth - added getLanguageNames() for use in ScriptModuleTypeProvider
  */
 public interface ScriptEngineFactory {
 
     /**
-     * @return the list of supported language endings e.g. py, jy
+     * @return the list of supported language endings, e.g. py, groovy, js
      */
     List<String> getLanguages();
```

Comment:
Sounds like the "file extensions" (WRT your `getLanguageNames()` comment).

Reply:
This method returns a list of file extensions _and_ MimeTypes. ScriptEngineManagerImpl takes the list and makes a Map out of it, adding in the ScriptEngines associated with each. I'll update the doc with more detail. 

getLanguageNames returns a Map with one name and one MimeType, for use in the ParameterOption when selecting a scripting language in the UI. I tried to combine these methods, but it was cleaner just having both.